app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt

```diff
@@ -229,3 +226,23 @@ object NotificationManager {
         return this
     }
 }
+
+/**
+ * A helper class that produces unique IDs for the notifications.
+ *
+ * The timestamp prevents collisions across app restarts, and the counter ensures uniqueness during
+ * high-frequency operations within the same millisecond. This method guarantees unique IDs without
+ * the need for persistent storage or complex logic.
+ */
+object UniqueIdGenerator {
```

Comment:
We have a solution for this already, let's use that instead?

https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/support/base/src/main/java/mozilla/components/support/base/ids/SharedIdsHelper.kt

General advice: if you come across a bug that requires a solution for that particular app, we should think, is this something we need to solve for all our apps and can we do that in a shared way? If not, then we should document the reasoning why.

Reply:
You sound like a true A-C builder ;)